File: Interviews/LeetCode/Topic-Wise/Strings/Medium/Integer-to-Roman/12.py
class Solution:
    def intToRoman(self, num: int) -> str:
        numerals, roman = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"], ""
        i, j, k, l = 0, 1000, 400, 100
        while i < 14:
            digitsReqd = int(num // j)
            for z in range(digitsReqd): roman += numerals[i]
            num %= j
            if i > 0 and i % 4 == 3:
                j -= (k - l)
                k, l = k / 10, l / 10
            else:
                j -= k if (i & 1) else l
            i += 1
        
        return roman

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-20/12.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<string> numerals = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        string roman = "";
        for (int i = 0, j = 1000, k = 400, l = 100; i < 14; i++) {
            int digitsReqd = num / j;
            while (digitsReqd--) {
                roman += numerals[i];
            }
            num %= j;
            if (i > 0 and i % 4 == 3) {
                j -= (k - l);
                k /= 10, l /= 10;
            } else {
                j -= (i & 1) ? k : l;
            }
        }

        return roman;
    }
};